presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateSetAggregation.java

```diff
@@ -31,53 +31,53 @@
+    public static final double DEFAULT_STANDARD_ERROR = 0.01625;
 
     private ApproximateSetAggregation() {}
 
-    public static HyperLogLog newHyperLogLog()
-    {
-        return HyperLogLog.newInstance(NUMBER_OF_BUCKETS);
-    }
-
     @InputFunction
     public static void input(@AggregationState HyperLogLogState state, @SqlType(StandardTypes.DOUBLE) double value)
     {
-        HyperLogLog hll = getOrCreateHyperLogLog(state);
-        state.addMemoryUsage(-hll.estimatedInMemorySize());
-        hll.add(Double.doubleToLongBits(value));
-        state.addMemoryUsage(hll.estimatedInMemorySize());
+        input(state, value, DEFAULT_STANDARD_ERROR);
     }
 
     @InputFunction
     @LiteralParameters("x")
     public static void input(@AggregationState HyperLogLogState state, @SqlType("varchar(x)") Slice value)
     {
-        HyperLogLog hll = getOrCreateHyperLogLog(state);
-        state.addMemoryUsage(-hll.estimatedInMemorySize());
-        hll.add(value);
-        state.addMemoryUsage(hll.estimatedInMemorySize());
+        input(state, value, DEFAULT_STANDARD_ERROR);
     }
 
     @InputFunction
     public static void input(@AggregationState HyperLogLogState state, @SqlType(StandardTypes.BIGINT) long value)
     {
-        HyperLogLog hll = getOrCreateHyperLogLog(state);
+        input(state, value, DEFAULT_STANDARD_ERROR);
+    }
+
+    @InputFunction
+    public static void input(@AggregationState HyperLogLogState state, @SqlType(StandardTypes.DOUBLE) double value, @SqlType(StandardTypes.DOUBLE) double maxStandardError)
+    {
+        input(state, Double.doubleToLongBits(value), maxStandardError);
+    }
+
+    @InputFunction
+    @LiteralParameters("x")
+    public static void input(@AggregationState HyperLogLogState state, @SqlType("varchar(x)") Slice value, @SqlType(StandardTypes.DOUBLE) double maxStandardError)
+    {
+        HyperLogLog hll = ApproximateCountDistinctAggregation.getOrCreateHyperLogLog(state, maxStandardError);
         state.addMemoryUsage(-hll.estimatedInMemorySize());
         hll.add(value);
         state.addMemoryUsage(hll.estimatedInMemorySize());
     }
 
-    private static HyperLogLog getOrCreateHyperLogLog(@AggregationState HyperLogLogState state)
+    @InputFunction
+    public static void input(@AggregationState HyperLogLogState state, @SqlType(StandardTypes.BIGINT) long value, @SqlType(StandardTypes.DOUBLE) double maxStandardError)
     {
-        HyperLogLog hll = state.getHyperLogLog();
-        if (hll == null) {
-            hll = newHyperLogLog();
-            state.setHyperLogLog(hll);
-            state.addMemoryUsage(hll.estimatedInMemorySize());
-        }
-        return hll;
+        HyperLogLog hll = ApproximateCountDistinctAggregation.getOrCreateHyperLogLog(state, maxStandardError);
+        state.addMemoryUsage(-hll.estimatedInMemorySize());
+        hll.add(value);
+        state.addMemoryUsage(hll.estimatedInMemorySize());
```

Comment:
It will throw an `IllegalArgumentException` in the combine method below, which will show up as an internal error in Presto (something that shouldn't happen), where this should show up as a user error.  I think it should be repackaged: `throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e.getMessage(), e);`